File: html/old/R2/SACO R2 2007 (Solutions)/dominoes.java
 //class dominoes
 //Problem "Dominoes" for SACO 2nd round 2007
 //Solution by Shen Tian
 
 import java.io.*;
 import java.util.*;
 
 public class dominoes {
   public static void main(String[] args) throws IOException{
     BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
     int N = Integer.parseInt(in.readLine());
     int[] a = new int[N];
     for (int i = 0; i < N; i++)
       a[i] = Integer.parseInt(in.readLine());
     int ans = 0;
     for (int i = 0; i < N; i++){
       boolean[] knocked = new boolean[N];
       Arrays.fill(knocked, false);
       int dom = i;
       int num = 0;
       while (!knocked[dom]){
         knocked[dom]=true;
         num++;
         dom = (dom + a[dom]) % N;
       }
       if (num==N)
         ans++;
     }
     System.out.println(ans);
   }
 }

File: html/old/backup/sacor2_2007/dominoes.py
#!/usr/bin/env python
# SACO Round 2 2007
# Solution by Marco Gallotta

N = int(raw_input())
A = []
for i in xrange(N):
    A.append(int(raw_input()))

ans = 0 
for dom in xrange(N):
    knocked = [False] * N 
    num = 0 
    while not knocked[dom]:
        knocked[dom] = True
        num += 1
        dom = (dom + A[dom]) % N 
    if num == N:
        ans += 1

print ans
